### ISSUE TITLE
Use JUnitRunner to run ScalaTests in parallel 

### ISSUE THREAD

**Issue by gerashegalov** [NVIDIA/spark-rapids#9787]
SparkQueryCompareTestSuite-based tests cannot be run in with `-Dparallel` using the standard ScalaTest runner because it only supports intra-JVM parallelism,  and the tests would clobber each other's Spark sessions.  

With the surefire plugin it is possible to run multiple tests across forked JVMs and if the thread count per JVM is one the SparkQueryCompareTestSuite-based test will work correctly. 

There are a couple of ways to to run ScalaTests in the JUnit framework. the least amount of change is by annotating Suite classes with
`@org.junit.runner.RunWith(classOf[org.scalatestplus.junit.JUnitRunner])` (easy to script / bulk edit in IDE)

With the prototype [branch](https://github.com/NVIDIA/spark-rapids/compare/branch-23.12...gerashegalov:junitRunner?expand=1) that does not take care of many things like automatically calculating memory as we do with xdist, setting up logging and reporting right, etc, we can see a positive effect

`branch-23.12` using ScalaTest runner
```bash
$ mvn package -pl tests -am -Dmaven.scalastyle.skip -Dmaven.scaladoc.skip -Dbuildver=330 \
      -Dsuffixes='.*(Orc|Parquet).*Suite.*'
Tests: succeeded 130, failed 0, canceled 24, ignored 9, pending 0
[INFO] RAPIDS Accelerator for Apache Spark Tests .......... SUCCESS [07:43 min]
```

junitRunner branch with the same set of tests
```bash
$ mvn package -pl tests -am -Dmaven.scalastyle.skip -Dmaven.scaladoc.skip -Dbuildver=330 -Dsuffixes=NONE \
      -Dsurefire.includes='%regex[.*(Orc|Parquet).*Suite.*]'
Tests run: 130, Failures: 0, Errors: 0, Skipped: 0
[INFO] RAPIDS Accelerator for Apache Spark Tests .......... SUCCESS [04:32 min]
```

**Comment by gerashegalov** [comment]
cc @pxLi

**Comment by pxLi** [comment]
> cc @pxLi

Looks promising! Please let us know when the tests are ready to go (good to have some easy-to-poke flag to change the forkCount in script), we can help with the resource planning and the CI-related updates later, thanks!

**Comment by gerashegalov** [comment]
I was hoping to get your help in taking it over the finish line.

**Comment by pxLi** [comment]
> I was hoping to get your help in taking it over the finish line.

Currently we may not have time for this one, 
cc @GaryShen2008 to help with the potential UT speed-up approach thanks

**Comment by GaryShen2008** [comment]
Postpone to 24.04

**Comment by pxLi** [comment]
bump up @yinqingh to help check when you have time, thanks
